### PR TITLE
Enhancement: Tab to be set from outside, Keyboard navigation

### LINKED ISSUE #738: Tabs: Set Active tab from the outside
**Is your feature request related to a problem? Please describe.**

In our use-case (Angular application) there is a `ifx-tabs` component. By using its `ifxTabIndex`-Event, we are routing to other pages. That means if the user clicks the second tab, we handle the event and use our router to navigate to a certain route, while the displayed-content is below the `ifx-tabs` itself in the `router-outlet`, like the following.

```Javascript
<ifx-tabs orientation="horizontal" (ifxTabIndex)="onSelectTab($event)">
      <ifx-tab header="..."></ifx-tab>
        ...
</ifx-tabs>
<router-outlet></router-outlet>
```

Regarding tabs I have the following issues:
- I am unable to provide a dynamic value for the `ifx-tab` header-property (like from a array of strings `[header]="tabs[0]"`)
- I can not set the active tab from the outside, controlled by a method. That means that in my case, if a user opens the application with a url which maps to the second tab, I can show the content in the `router-outlet`, but not set the active tab accordingly, like in our approach

**Describe the solution you'd like**

A `@method` annotated to set the active tab (maybe the already included `setActiveTab` will do)

### PR DESCRIPTION
**Pull request by verena-ifx**
By creating this pull request you agree to the terms in CONTRIBUTING.md.
https://github.com/Infineon/.github/blob/master/CONTRIBUTING.md
--- DO NOT DELETE ANYTHING ABOVE THIS LINE ---

CONTRIBUTING.md also tells you what to expect in the PR process.

Description
Tab index can now be set from the outside;
Removed unnecessary code and added the updated Tab component to the Angular+React+Vue example apps
Keyboard navigation added

<details>
  <summary>📦 Published PR as canary version: <code>20.37.0--canary.741.683712c531c460cfe5ac518ff3c3bae7a5eec2c0.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @infineon/infineon-design-system-stencil@20.37.0--canary.741.683712c531c460cfe5ac518ff3c3bae7a5eec2c0.0
  # or 
  yarn add @infineon/infineon-design-system-stencil@20.37.0--canary.741.683712c531c460cfe5ac518ff3c3bae7a5eec2c0.0
  ```
</details>

### CODE REVIEW

**Comment by MichaelHolley** [comment]
Checked out the canary-build and as far as I can tell, the dynamic header-values and setActiveTab improvements work as expected. Looking forward to its merge 😃

**Comment by tishoyanchev** [comment]
Is this PR done?

**Comment by tishoyanchev** [comment]
General note:
1) Let's avoid making overarching changes inside a PR dedicated to one particular issue. From the title, it seems like this PR is about the tab issue, and the keyboard navigation therefore refers only to the tabs, and yet changes are made to all components regarding keyboard navigation. I think these changes should have been separate, and not inside this PR. 
2) Let's do more detailed descriptions. Not super detailed, but detailed enough so as to be clear by reading it what exactly is being changed and why.

File: packages/components/src/components/tabs/tabs.tsx [hunk 1]
```diff
@@ -52,38 +67,31 @@ export class IfxTabs {
...
     const tabs = this.el.querySelectorAll('ifx-tab');
-    this.tabTitles = Array.from(tabs).map((tab) => tab.getAttribute('header'));
-    this.disabledTabs = Array.from(tabs).map((tab) => tab.getAttribute('disabled'));
-
-    this.tabObjects = this.tabTitles.map((header, index) => {
+    this.tabObjects = Array.from(tabs).map((tab) => {
       return {
-        header: header,
-        disabled: (this.disabledTabs[index] === 'true') // set disabled property based on the corresponding value in the disabledTabs array
+        header: tab?.header === null ? tab?.getAttribute('header') !== null : tab.header,
```
Comment (tishoyanchev) [inline packages/components/src/components/tabs/tabs.tsx:93]: Not sure I understand this.. 
If header is null, we then check if null !== null? 🤔

File: packages/components/src/components/checkbox/checkbox.tsx [hunk 2]
```diff
@@ -23,18 +23,20 @@ export class Checkbox {
     if (!this.disabled) {
       if (this.inputElement.indeterminate) {
         this.internalValue = true;
+        this.value = this.internalValue;
         this.indeterminate = false;
       } else {
         this.internalValue = !this.internalValue;
+        this.value = this.internalValue;
```
[the remark this answers is not in the document]
Comment (verena-ifx) [reply packages/components/src/components/checkbox/checkbox.tsx:30]: both need to be in sync, this is why.

**Comment by verena-ifx** [comment]
> General note:
> 
> 1. Let's avoid making overarching changes inside a PR dedicated to one particular issue. From the title, it seems like this PR is about the tab issue, and the keyboard navigation therefore refers only to the tabs, and yet changes are made to all components regarding keyboard navigation. Besides that, there are also changes to the scss. I think these changes should have been separate, and not inside this PR.
> 2. Let's do more detailed descriptions. Not super detailed, but detailed enough so as to be clear by reading it what exactly is being changed and why.

changes to the other components are only r

> General note:
> 
> 1. Let's avoid making overarching changes inside a PR dedicated to one particular issue. From the title, it seems like this PR is about the tab issue, and the keyboard navigation therefore refers only to the tabs, and yet changes are made to all components regarding keyboard navigation. Besides that, there are also changes to the scss. I think these changes should have been separate, and not inside this PR.
> 2. Let's do more detailed descriptions. Not super detailed, but detailed enough so as to be clear by reading it what exactly is being changed and why.

the majority of the other changes are simply replacing hard-coded tokens, but yes generally you are right, this should be done separately.

**Comment by verena-ifx** [comment]
> > General note:
> > 
> > 1. Let's avoid making overarching changes inside a PR dedicated to one particular issue. From the title, it seems like this PR is about the tab issue, and the keyboard navigation therefore refers only to the tabs, and yet changes are made to all components regarding keyboard navigation. Besides that, there are also changes to the scss. I think these changes should have been separate, and not inside this PR.
> > 2. Let's do more detailed descriptions. Not super detailed, but detailed enough so as to be clear by reading it what exactly is being changed and why.
> 
> 
> > General note:
> > 
> > 1. Let's avoid making overarching changes inside a PR dedicated to one particular issue. From the title, it seems like this PR is about the tab issue, and the keyboard navigation therefore refers only to the tabs, and yet changes are made to all components regarding keyboard navigation. Besides that, there are also changes to the scss. I think these changes should have been separate, and not inside this PR.
> > 2. Let's do more detailed descriptions. Not super detailed, but detailed enough so as to be clear by reading it what exactly is being changed and why.
> 
> the majority of the other changes are simply replacing hard-coded tokens, but yes generally you are right, this should be done separately.

File: packages/components/src/components/tabs/tabs.tsx [hunk 3]
```diff
@@ -52,38 +67,31 @@ export class IfxTabs {
   // needed for smooth border transition
   reRenderBorder() {
     const borderElement = this.el.shadowRoot.querySelector('.active-border') as HTMLElement;
-    if (borderElement && this.tabHeaderRefs[this.activeTabIndex]) {
+    if (borderElement && this.tabHeaderRefs[this.internalActiveTabIndex]) {
       if (this.orientation === 'horizontal') {
```
[the remark this answers is not in the document]
Comment (verena-ifx) [reply packages/components/src/components/tabs/tabs.tsx:71]: the orientation? didnt make changes to that and its configured on storybook via 2 options only. But we can add the lowerCase check.
Or do you mean the tabIndex? thats passed as a number, so shouldnt matter

File: packages/components/src/components/checkbox/checkbox.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (tishoyanchev) [reply packages/components/src/components/checkbox/checkbox.tsx:30]: But if you need to reassign the **value** to the **internalValue**, then the **value** Prop has been changed externally, and therefore both will be synced by assigning the **value** to the **internalValue**.

File: packages/components/src/components/tabs/tabs.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (tishoyanchev) [reply packages/components/src/components/tabs/tabs.tsx:69]: Yes, I was referring to the **orientation** prop. I can see people typing **Horizontal** with capital letter, and wondering why is it not working.

File: examples/wrapper-components/react-javascript/src/components/Tabs/Tabs.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+  const handleChange = (event) => {
+    console.log("emitting active tab index: ", event.detail)
+  };
+
+
+  const setTab = () => {
+    const next = Math.floor(Math.random() * (3));
+    console.log("set next active tab: ", next)
+    tabsElement.current?.setActiveTab(next);
```
Comment (tishoyanchev) [inline examples/wrapper-components/react-javascript/src/components/Tabs/Tabs.js:24]: An error occurs here:
**_tabsElement$current.setActiveTab is not a function**

File: examples/wrapper-components/vue-javascript/src/components/Tabs.vue [hunk 5]
```diff
@@ -3,22 +3,60 @@
...
+
+function handleChange(event) {
+  console.log("emitting active tab index: ", event.detail)
+};
+
+function setTab() {
+  const next = Math.floor(Math.random() * (3));
+  console.log("set next active tab: ", next)
+  if (tabsElement.value) {
+    tabsElement.value.$el.setActiveTab(next);
```
Comment (tishoyanchev) [inline examples/wrapper-components/vue-javascript/src/components/Tabs.vue:56]: Like in React, there is an error here:
**Cannot read properties of undefined (reading 'setActiveTab')**

File: packages/components/src/components/tabs/tabs.tsx [hunk 6]
```diff
@@ -9,87 +9,66 @@ import { Component, h, Prop, State, Element, Listen, Watch, Event, EventEmitter
...
+    this.tabObjects = Array.from(tabs).map((tab) => {
       return {
-        header: header,
-        disabled: (this.disabledTabs[index] === 'true') // set disabled property based on the corresponding value in the disabledTabs array
+        header: tab?.header === null ? tab?.getAttribute('header') !== null : tab.header,
+        disabled: tab?.disabled === true // (tab.getAttribute('disabled') === 'true')
       }
     });
 
     this.tabRefs = Array.from(tabs);
```
[the remark this answers is not in the document]
Comment (verena-ifx) [reply packages/components/src/components/tabs/tabs.tsx:68]: one is the actual content, the other are the html references

File: packages/components/src/components/tabs/tabs.tsx [hunk 7]
```diff
@@ -9,87 +9,66 @@ import { Component, h, Prop, State, Element, Listen, Watch, Event, EventEmitter
...
-    const event = new CustomEvent(eventName, {
-      bubbles: true,
-      composed: true,
-    });
-    element.dispatchEvent(event);
+  @Watch('activeTabIndex')
+  onActiveTabIndexChange(newIndex: number, oldIndex: number) {
+    this.internalPrevActiveTabIndex = oldIndex;
+    this.internalActiveTabIndex = this.tabObjects[newIndex]?.disabled ? oldIndex : newIndex;
+    this.activeTabIndex = this.internalActiveTabIndex;
```
[the remark this answers is not in the document]
Comment (verena-ifx) [reply packages/components/src/components/tabs/tabs.tsx:30]: it is (the activetabindex is passed here as the newIndex, but it also needs to be updated again itself

**Comment by verena-ifx** [comment]
> 1. On first page load, tab 1 is the first tab, tab 2 is the second tab, tab 3 is the content. But after that, if you click away, and then click tab key on the keyboard again, tab 1 is the first tab, tab 2 is the content. The tab 2 never gets focused again, not even with key right.
> 2. Focus not working on React Javascript
> 3. Cannot run on Vue-Javascript. Getting an error.

the keyboard navigation is not working with key navigation anymore, only with tabs. Focus is not working, I told you already

File: packages/components/src/components/tabs/tab.tsx [hunk 8]
```diff
@@ -7,11 +7,8 @@ import { Component, h, Element } from '@stencil/core';
 })
 export class IfxTab {
   @Element() el: HTMLElement;
-  //@Prop() header: string;
-  //@Prop() disabled: boolean;
-
-  //@Event({ eventName: 'tabBecameActive' }) tabBecameActive: EventEmitter;
-  //@Event({ eventName: 'tabBecameInactive' }) tabBecameInactive: EventEmitter;
+  @Prop() header: string;
+  @Prop() disabled: boolean;
```
[the remark this answers is not in the document]
Comment (tishoyanchev) [reply packages/components/src/components/tabs/tab.tsx:11]: Actually the 2 props are being used inside the tabs component to check if that particular tab is disabled, and what is its header. Only the HTMLElement is not needed.

**Pull request merged**